sound/soc/ti/davinci-mcasp.c

```diff
@@ -2111,8 +2112,11 @@ static int davinci_mcasp_gpio_request(struct gpio_chip *chip, unsigned offset)
 	}
 
 	/* Do not change the PIN yet */
+	ret = pm_runtime_resume_and_get(mcasp->dev);
+	if (ret < 0 && ret != -EACCES)
+		return ret;
```

Comment:
Please do not hide the `-EACCES`, it meas that the the device remained _disabled_ because the `rpm_resume()` did not run because the disable_depth was different from 0.
We can only return 0 if the McASP _is_ enabled.